### PR TITLE
Add ARCH=arc support for docker builds

### PR DESCRIPTION
**Pull request by khilman**

### CODE REVIEW

**Comment by khilman** [comment]
Testing for gcc-7:
```
$ docker build -t kci-arc-gcc7 .
Sending build context to Docker daemon  2.048kB
Step 1/4 : FROM kernelci/build-base
 ---> 6499b031c72d
Step 2/4 : ENV CROSS_TARBALL arc_gnu_2017.09_prebuilt_uclibc_le_archs_linux_install.tar.gz
 ---> Using cache
 ---> 9606cf54b3d8
Step 3/4 : ENV CROSS_URL https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/download/arc-2017.09-release/${CROSS_TARBALL}
 ---> Using cache
 ---> 46f6855f8abe
Step 4/4 : RUN cd /tmp; wget -q ${CROSS_URL}; tar -C /usr --strip-components=1 -xaf ${CROSS_TARBALL}; rm -f ${CROSS_TARBALL}
 ---> Using cache
 ---> d3c7897479a6
Successfully built d3c7897479a6
Successfully tagged kci-arc-gcc7:latest

$ docker run -it kci-arc-gcc7 arc-linux-gcc -v
^[[B^[[B^[[B^[[B^[[B^[[B^[[BUsing built-in specs.
COLLECT_GCC=arc-linux-gcc
COLLECT_LTO_WRAPPER=/usr/bin/../libexec/gcc/arc-snps-linux-uclibc/7.1.1/lto-wrapper
Target: arc-snps-linux-uclibc
Configured with: /SCRATCH/arcjenkins/custom-ru20-arcgnu2/workspace/arc_gnu_toolchain_release/gcc/configure --target=arc-snps-linux-uclibc --with-cpu=hs38 --disable-multilib --with-pkgversion='ARCv2 ISA Linux uClibc toolchain 2017.09' --with-bugurl=https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/issues --enable-fast-install=N/A --with-endian=little --disable-werror --enable-languages=c,c++ --prefix=/SCRATCH/arcjenkins/custom-ru20-arcgnu2/workspace/arc_gnu_toolchain_release/toolchain/../release_output/arc_gnu_2017.09_prebuilt_uclibc_le_archs_linux_install --enable-shared --without-newlib --disable-libgomp --with-gnu-as --with-gnu-ld --with-python=no --with-sysroot=/SCRATCH/arcjenkins/custom-ru20-arcgnu2/workspace/arc_gnu_toolchain_release/toolchain/../release_output/arc_gnu_2017.09_prebuilt_uclibc_le_archs_linux_install/arc-snps-linux-uclibc/sysroot
Thread model: posix
gcc version 7.1.1 20170710 (ARCv2 ISA Linux uClibc toolchain 2017.09) 
```

**Comment by khilman** [comment]
Testing for gcc-8:

```
$ docker build -t kci-arc-gcc8 .
Sending build context to Docker daemon  2.048kB
Step 1/4 : FROM kernelci/build-base
 ---> 6499b031c72d
Step 2/4 : ENV CROSS_TARBALL arc_gnu_2018.09-rc1_prebuilt_glibc_le_archs_linux_install.tar.gz
 ---> Using cache
 ---> fed2d4e7bd82
Step 3/4 : ENV CROSS_URL https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/download/arc-2018.09-rc1/${CROSS_TARBALL}
 ---> Using cache
 ---> c7be1e805c83
Step 4/4 : RUN cd /tmp; wget -q ${CROSS_URL}; tar -C /usr --strip-components=1 -xaf ${CROSS_TARBALL}; rm -f ${CROSS_TARBALL}
 ---> Using cache
 ---> 0603c6192863
Successfully built 0603c6192863
Successfully tagged kci-arc-gcc8:latest

$ docker run -it kci-arc-gcc8 arc-linux-gcc -v
Using built-in specs.
COLLECT_GCC=arc-linux-gcc
COLLECT_LTO_WRAPPER=/usr/bin/../libexec/gcc/arc-snps-linux-gnu/8.2.1/lto-wrapper
Target: arc-snps-linux-gnu
Configured with: /SCRATCH/arcjenkins/custom-ru20-arcgnu2/workspace/arc_gnu_toolchain_release/gcc/configure --target=arc-snps-linux-gnu --with-cpu=hs38 --disable-multilib --with-pkgversion='ARC HS GNU/Linux glibc toolchain 2018.09-rc1' --with-bugurl=https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/issues --enable-fast-install=N/A --with-endian=little --disable-werror --enable-languages=c,c++ --prefix=/SCRATCH/arcjenkins/custom-ru20-arcgnu2/workspace/arc_gnu_toolchain_release/toolchain/../release_output/arc_gnu_2018.09-rc1_prebuilt_glibc_le_archs_linux_install --enable-shared --without-newlib --disable-libgomp --with-gnu-as --with-gnu-ld --with-python=no --with-sysroot=/SCRATCH/arcjenkins/custom-ru20-arcgnu2/workspace/arc_gnu_toolchain_release/toolchain/../release_output/arc_gnu_2018.09-rc1_prebuilt_glibc_le_archs_linux_install/arc-snps-linux-gnu/sysroot
Thread model: posix
gcc version 8.2.1 20180814 (ARC HS GNU/Linux glibc toolchain 2018.09-rc1) 
```

**Comment by khilman** [comment]
@gctucker this should fix the ARCH=arc builds.  
TL;DR; the current toolchain only supports a subset of the available defconfigs, so either we filter them out (as proposed in third patch here) or we just accept a bunch of failed builds (including the default 'make defconfig')

**Comment by mattface** [comment]
I think the build emails are noisy enough as it is, we probably don't want to add known failing ones just yet. 
Maybe only add all the failing ones to an ARC developer tree?

**Comment by mattface** [comment]
Also, are the ARC people going to sponsor a builder? :P

**Comment by khilman** [comment]
The ARC developers won't be interested in the failures either, because they are expected to fail when you use the wrong toolchain.

So for now, I suggest we just merge this as is.

**Comment by gctucker** [comment]
We still need to get a successful build at least with `nsim_hs_defconfig` on staging.  I'll give it another spin with this version.

File: jenkins/dockerfiles/gcc-8_arc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM kernelci/build-base
+
+# 
+# pre-built toolchains for ARC available here:
+#    https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/
+#
+ENV CROSS_TARBALL arc_gnu_2018.09-rc1_prebuilt_glibc_le_archs_linux_install.tar.gz
```
[the remark this answers is not in the document]
Comment (khilman) [reply jenkins/dockerfiles/gcc-8_arc/Dockerfile]: That seems to be the case today by cooincedence, but I don't think that's a naming convention.

**Comment by khilman** [comment]
Fixed the additional curly brace.

File: jenkins/dockerfiles/gcc-8_arc/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (gctucker) [reply jenkins/dockerfiles/gcc-8_arc/Dockerfile]: OK - it just means one may need to extract the tarball to find out.  But it's fine once it's in a Docker image - just being curious.

**Comment by gctucker** [comment]
Nitpick: I also added a trailing `,` in build.py cross_compilers dict...

**Comment by khilman** [comment]
yes, thanks for the cleanup.
